tag_utils/cli/tag_cleaner.py

```diff
@@ -51,13 +51,14 @@ def tag_cleaner(args):
 
     exclude_components = set(args.exclude.split(','))
 
-    if '-candidate' in args.brew_tag:
-        base_tag = args.brew_tag[0:-len('-candidate')]
-    elif '-pending' in args.brew_tag:
-        base_tag = args.brew_tag[0:-len('-pending')]
-    elif '-override' in args.brew_tag:
-        base_tag = args.brew_tag[0:-len('-override')]
-    else:
+    base_tag = None
+    for sub_tag in ['-trunk-candidate', '-trunk-override', '-candidate',
```

Comment:
On change that could avoid the looping is to use a regex for example:
`result = re.match(r'(.+)-(trunk-candidate|trunk-override|candidate|pending|override)', args.brew_tag)`

Then you can check for `None` and use the `result.group[1]` for the base_tag and `result.group[2]` for the sub_tag if you need it.

Reply:
have to make sure it does right order of precedence but sure.

Just changes looping for regex, but then have to verify regex does the right thing.

Reply:
`
>>> result = re.match(r'(.+)-(trunk-candidate|trunk-override|candidate|pending|override)', 'foo-trunk-candidate')
>>> result.groups()
('foo-trunk', 'candidate')
`
Hmm would need to get it to be greedy when doing the searching for that approach to work.

Reply:
After doing some testing, it seems we don't have a stable pattern so a regex is not going to work. Using the looping seems to be the best route.